climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
+
+    Returns:
+        float (CDF at z)
+    """
+    c = np.cos(theta)
+    s = np.sin(theta)
+    if z >= np.sqrt(3) * (c + s):
+        res = 1
+    elif z > np.sqrt(3) * (c - s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (-3 - z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 9 * np.sin(2 * theta))
+    elif z > np.sqrt(3) * (-c + s):
+        res = 1 / 6 * (3 + np.sqrt(3) * z / c)
+    elif z > -np.sqrt(3) * (c + s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 3 * (1 + np.sin(2 * theta)))
+    else:
+        res = 0
+    return res
+
+
+@jit
+def _get_bearing_angle(lon, lat):
+    """Compute bearing angle of great circle paths defined by consecutive points
+
+    Returns initial bearing (also called forward azimuth) of the n-1 great circle
+    paths define by n consecutive longitude/latitude points. The bearing is the angle
+    (clockwise from North) which if followed in a straight line along a great-circle
+    arc will take you from the start point to the end point. See also:
+    http://www.movable-type.co.uk/scripts/latlong.html
+    Here, the bearing of each pair of consecutive points is computed.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        bearing_angle (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lon_1 = lon[:-1]
+    lat_2 = lat[1:]
+    lon_2 = lon[1:]
+    delta_lon = lon_2 - lon_1
+    # what to do with the points that don't move?
+    #   i.e. where lat_2=lat_1 and lon_2=lon_1? The angle does not matter in
+    # that case because angular distance will be 0.
+    earth_ang_fix = np.arctan2(np.sin(delta_lon) * np.cos(lat_2),
+                               np.cos(lat_1) * np.sin(lat_2) - \
+                               np.sin(lat_1) * np.cos(lat_2) * np.cos(delta_lon))
+    return np.degrees(earth_ang_fix)
+
+
+@jit
+def _get_angular_distance(lon, lat):
+    """Compute the angular distance of great circle paths defined by consecutive points
+
+    Uses the haversine formulat to get the great circle distance between each pair
+    of consecutive points, in decimal degrees.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        Angular distance (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
+
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lat_2 = lat[1:]
+    delta_lon = lon[1:] - lon[:-1]
+    delta_lat = lat_2 - lat_1
+    a = np.sin(delta_lat / 2.0) ** 2 + np.cos(lat_1) * np.cos(lat_2) * np.sin(delta_lon / 2.0) ** 2
+    c = np.degrees(2 * np.arctan2(np.sqrt(a), np.sqrt(1 - a)))
```

Comment:
Is this equivalent to the formula from here: https://en.wikipedia.org/wiki/Haversine_formula ?

Why to you assume circles of radius 1?

Reply:
I am using distance in radians, not in km, hence it looks like I'm using a radius of 1. But this is consistent throughout the code so all good. I preferred that to relying on an assumption about Earth's radius, also because it is then clear it assumed a sphere.

Reply:
I think it is necessary to explain this in a short comment.

Reply:
To me the term 'angular distance' is clear: it is expressing a distance on a sphere by its angle from the sphere center. So if this angle is in radians, it would happen to be the distance for a sphere if radius 1, but actually it is an angle, not a distance per se. If you can be more specific on what you would add there as a comment I can include that of course.